[4단계 - 동시성 확장하기] 레오(김준영) 미션 제출합니다.

kjyyjk opened the pull request:
안녕하세요 대니~
마지막 step도 리뷰 잘부탁드려요.

이전 pr에서 남겨주신 질문에 대한 답변입니다.
https://github.com/woowacourse/java-http/pull/1038#discussion_r2349495945

kjyyjk committed:
feat: 4-1 Executors로 Thread Pool 적용

kjyyjk committed:
feat: 4-1 Executors로 Thread Pool 적용

kjyyjk committed:
feat: 4-2 동시성 컬렉션 사용하기

goohong commented on tomcat/src/main/java/org/apache/catalina/connector/Connector.java:
```diff
@@ -66,8 +71,7 @@ private void process(final Socket connection) {
         if (connection == null) {
             return;
         }
-        var processor = new Http11Processor(connection);
-        new Thread(processor).start();
+        executorService.submit(new Http11Processor(connection));
```
`execute()`가 아닌 `submit()`을 사용하신 이유가 있나요?
반환 값을 사용하지 않는 것 같아서요!

goohong reviewed (changes requested):
안녕하세요 레오~
궁금한 점 하나 코멘트로 남겼습니다 👍

kjyyjk committed:
refactor: excutorService.execute 사용

- ExecutorService#submit과 execute는 모두 비동기 병렬 처리

- submit은 결과가 담긴 Future를 반환. 예외 처리 등 결과가 필요한 경우 유용.

- execute는 반환 타입이 void. fire and forget 방식.

goohong reviewed (approved):
미션 수고하셨습니다 👍
